lib/vmdb/settings/validator.rb

```diff
@@ -31,6 +31,23 @@ def validate
 
       private
 
+      def integer_with_method?(num)
+        if is_integer?(num.to_i_with_method) && num.present?
+          # handling the "Five".to_i == 0 case
+          #
+          # Allows:
+          #
+          #   - 0
+          #   - 0.minutes (always will be zero)
+          #   - any integer
+          #   - any integer + method
+          #
+          num.to_i_with_method != 0 || num.to_s.match?(/^0(?=\..*|$)/)
+        else
+          num.blank?
+        end
+      end
```

Comment:
I think this can be

```suggestion
      def integer_with_method?(num)
        is_integer?(num) || num.number_with_method?
      end
```

I wasn't sure what the blank case is convering, but the above handles nil and `""` as well.

Note that `to_i_with_method` works with numbers that don't have a method either.  It is a `to_i` method method, but with specialization for methods:

```ruby
nil.number_with_method? #  => false
"".number_with_method?  #  => nil

"5".number_with_method? #  => false
"5".to_i_with_method    # => 5

"5.seconds".number_with_method? #  => true
"5.seconds".to_i_with_method    # => 5

Reply:
Your suggestion fails the tests I had when I added it inline.

Reply:
Might be able to use it modified a bit, but would have to dig deeper as to why it fails.

Reply:
Fixed as well.

Just did the checks inline since after fixing the bad test cases, it made the method simpler and now unnecessary.